app/models/fs/message.rb

```diff
@@ -0,0 +1,133 @@
+    Message.create(
+      uuid: SecureRandom.uuid,
+      title: raw_message['message_type_name'],
+      sender_name: FS_SUBJECT_NAME,
+      recipient_name: raw_message['subject'],
+      delivered_at: Time.parse(raw_message['created_at']).getlocal,
+      replyable: false,
+      collapsed: collapsed?,
+      outbox: false,
+      metadata: {
+        "fs_message_id": raw_message['message_id'],
+        "fs_sent_message_id": raw_message['sent_message_id'],
+        "fs_status": raw_message['status'],
+        "fs_submitting_subject": raw_message['submitting_subject'],
+        "fs_submission_status": raw_message['submission_status'],
+        "fs_submission_created_at": Time.parse(raw_message['submission_created_at']).getlocal,
+        "dic": raw_message['dic']
+      },
+    )
+  end
+
+  def self.create_outbox_message(raw_message, associated_message_draft: nil)
+    Message.create(
+      uuid: (associated_message_draft.uuid if associated_message_draft) || SecureRandom.uuid,
+      title: raw_message['submission_type_name'],
+      sender_name: raw_message['subject'],
+      recipient_name: FS_SUBJECT_NAME,
+      delivered_at: Time.parse(raw_message['created_at']).getlocal,
+      replyable: false,
+      collapsed: collapsed?,
+      outbox: true,
+      metadata: {
+        "fs_form_id": (associated_message_draft.metadata['fs_form_id'] if associated_message_draft) || Fs::Form.where("identifier LIKE '#{raw_message['submission_type_id']}_%'")&.take&.id,
+        "fs_message_id": raw_message['message_id'],
+        "fs_status": raw_message['status'],
+        "submitting_subject": raw_message['submitting_subject'],
+        "dic": raw_message['dic']
+      },
+    )
+  end
+
+  def self.create_message_objects(message, raw_message)
+    raw_message.dig('message_container', 'objects').each do |raw_object|
+      tags = raw_object["signed"] ? [message.thread.box.tenant.signed_externally_tag!] : []
+
+      message_object = message.objects.create!(
+        # uuid: raw_object["id"], TODO uncomment when GO-130 is closed
+        is_signed: raw_object["signed"],
+        mimetype: raw_object["mime_type"],
+        name: raw_object["name"],
+        object_type: raw_object["class"],
+        tags: tags
+      )
+
+      if raw_object["encoding"] == "Base64"
+        message_object_content = Base64.decode64(raw_object["content"])
+      else
+        message_object_content = raw_object["content"]
+      end
+
+      MessageObjectDatum.create!(
+        blob: message_object_content,
+        message_object: message_object
+      )
+    end
+  end
+
+  def self.update_html_visualization(message)
+    message.update(
+      html_visualization: Fs::MessageHelper.build_html_visualization(message)
+    )
+
+    message.form_object&.update(
+      visualizable: message.html_visualization.present?
+    )
+  end
+
+  def self.collapsed?
+    # TODO urcit podmienky: odoslana sprava s potvrdenkou by mohla byt collapsed
```

Comment:
Kedze ta dorucenka je extremne dolezita pre nich, tak skor prave naopak.

Reply:
Je to tu zle naformulovane, mala som na mysli, ze podanie by mohlo byt collapsed v pripade, ze uz prisla dorucenka. Toto su zatial rychle komentare v ramci draftu, doladim este.